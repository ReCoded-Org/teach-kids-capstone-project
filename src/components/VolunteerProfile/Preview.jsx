// import React from 'react'

// export const Preview = ({files}) => {

//   if(files.length){
//     return files.map(file => 
//       {
//         return (<img src={`//localhost:5000/${file.filename}`} alt={file.originalname} />)
//       })
//   }

//   if (!files.length){
//     return null;
//   }

  
// }


//             {/* {files.map(file => <img src={`//localhost:5000/${file.filename}`} alt={file.originalname} />)} */}


