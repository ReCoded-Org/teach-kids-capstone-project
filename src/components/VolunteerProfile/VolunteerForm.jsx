import axios from "axios";
import React, {useState} from "react";

export default function VolunteerForm(props) {
  const [showModal, setShowModal] = React.useState(false);


  return (
    <>
      
      <div className=' flex flex-wrap justify-end py-2 mr-5'>
        <div className="basis-8/10"></div>
        <button 
          className="flex justify-end bg-customGreen text-white active:bh-customGreen font-bold uppercase text-sm px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
          type="button"
          onClick={() => setShowModal(true)}
      >
        Edit Profile
      </button>        
      </div>
      {showModal ? (
        <>
          <div
            className="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
          >
            <div className="relative w-auto my-6 mx-auto max-w-3xl">
              {/*content*/}
              <div className="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none">
                {/*header*/}
                <div className="flex items-start justify-between p-5 border-b border-solid border-slate-200 rounded-t">
                  <h3 className="text-3xl font-semibold">
                    Edit Pofile
                  </h3>
                  <button
                    className="p-1 ml-auto bg-transparent border-0 text-black opacity-50 float-right text-3xl leading-none font-semibold outline-none focus:outline-none"
                    onClick={() => setShowModal(false)}
                  >
                    <span className="bg-transparent text-black h-6 w-6 text-2xl block outline-none focus:outline-none">
                      X
                    </span>
                  </button>
                </div>
                {/*body*/}
                <div className="relative p-6 flex-auto">
                  {/* START OF FORM */}



                  <form className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4">

                  <div className="mb-4">
                      <label className="block text-gray-700 text-sm font-bold mb-2">
                        Select a Profile Photo
                      </label>
                      <input className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" type="file" placeholder="" />
                    </div>

                  <div className="mb-4">
                      <label className="block text-gray-700 text-sm font-bold mb-2">
                        Name
                      </label>
                      <input onChange={props.handleChange} name="name" className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" type="text" placeholder="Name" />
                    </div>

                    <div className="mb-4">
                      <label className="block text-gray-700 text-sm font-bold mb-2">
                        Description
                      </label>
                      <input onChange={props.handleChange} name="description" className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" type="text" placeholder="Description" />
                    </div>

                    <div className="mb-4">
                      <label className="block text-gray-700 text-sm font-bold mb-2">
                        Email
                      </label>
                      <input onChange={props.handleChange} name="email" className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" type="text" placeholder="Email" />
                    </div>

                    <div className="mb-4">
                      <label className="block text-gray-700 text-sm font-bold mb-2">
                        Phone Number
                      </label>
                      <input onChange={props.handleChange} name="phone" className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline" type="tel" placeholder="Phone Number" />
                    </div>


                    
                    {/* <input onChange={(event)=> setImageSelected(event.target.files[0])} className="w-100" type='file' /> <br></br> */}
                  </form>
                  {/* END OF FORM */}




                </div>
                {/*footer of the form*/}
                <div className="flex items-center justify-end p-6 border-t border-solid border-slate-200 rounded-b">
                  <button
                    className="text-red-500 background-transparent font-bold uppercase px-6 py-2 text-sm outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
                    type="button"
                    onClick={() => setShowModal(false)}
                  >
                    Close
                  </button>
                  <button
                    className="bg-customGreen text-white active:bg-emerald-600 font-bold uppercase text-sm px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
                    type="button"
                    onClick={() => {
                      setShowModal(false)
                      props.dataSender()
                    }}
                    
                  >
                    Save Changes
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div className="opacity-25 fixed inset-0 z-40 bg-black"></div>
        </>
      ) : null}
    </>
  );
}
