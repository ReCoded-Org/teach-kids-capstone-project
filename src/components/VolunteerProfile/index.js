import { toast } from 'react-toastify';


// started
const express = require('express')
const app= express();

const userUpload = require('./routes/index');

app.use((req, res, next) => {
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader(
        'Access-Control-Allow-Headers',
        'Origin, X-Requested-With, Content-Type, Accept, Authorization'
    );
    res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PATCH, DELETE');

    next();
});

app.use('/user', userUpload)


app.listen(3002, ()=> {
    console.log('Server is Running..')
})
// ended