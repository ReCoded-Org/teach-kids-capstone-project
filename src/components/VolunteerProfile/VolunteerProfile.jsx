import React, { useState } from "react";
import VolunteerForm from "./VolunteerForm";
import Axios from "axios";
import SubmitConfirm from "./SubmitConfirm";

import { getEvents } from "../../services/events.js";
import { useQuery } from "@tanstack/react-query";

// import { ToastContainer } from 'react-toastify';44

function VolunteerProfile({userId}) {
    const [info, setInfo] = useState([]); //changed {} to []
    const [file, setFile] = useState(null) // for uploadig the CV
    const [events, setEvents]= useState([]) // for events related to volunteer
    const [updated, setUpdated] = useState(false);
    const [showModal, setShowModal] = React.useState(false);

    function handleChange(e) {
        setInfo({ ...info, [e.target.name]: e.target.value });
    }

    function dataSender(e) {
        Axios.put(`https://reach-capstone.herokuapp.com/api/volunteers/${userId}`, info).then(
            (response) => {
                // console.log(response.data)
                setInfo(() => response.data);
                // console.log('info after dataSender',info)
                setUpdated(true);
                setTimeout(() => setUpdated(false), 2000);
            }
        );
    }


    // Upload Functions START:

    function onInputChange(e) {
        // console.log(e.target.value);
        // console.log(e.target.files);
        setFile(e.target.files[0])
    }

    function handleSubmit(e) {
        e.preventDefault();

        const data = new FormData()

        data.append('file', file)

        // For Uploaded File Preview (should be called before any loops or if conditions)
        // function onSuccess(savedFiles) {
        //     setFiles(savedFiles)
        // }
        // end

        Axios.patch('https://reach-capstone.herokuapp.com/api/profile', data)
            .then((response)=> {
                alert('Success') // he adds the toast here and below 

            })
            .catch((e) => {
                alert('Error', e)
            })

    }

    // END

   

    React.useEffect(() => {
        Axios.get(`https://reach-capstone.herokuapp.com/api/volunteers/${userId}`)
            .then((res) => {
                setInfo(res.data);
                // console.log('info after useEffect', info)
            })
            .catch((err) => {
                console.error("error in getting volunteer info", err);
            });
    }, [showModal]);

    React.useEffect(()=> {
        Axios.get(`https://reach-capstone.herokuapp.com/api/volunteers/${userId}`)
        .then((res)=> {
            setEvents(res.data)
        })
        .catch((err) => {
            console.error("error for applied events", err);
        });
    }, [])

    const { isLoading, error, data } = useQuery(["events"], () => getEvents());

    if (isLoading) return "Loading...";
    if (error) return "An error has occurred: " + error.message;

    
    return (
        <div className='bg-gray font-quicksand'>
            <VolunteerForm
                handleChange={handleChange}
                dataSender={dataSender}
                showModal={showModal}
                setShowModal={setShowModal}
            />

            {updated ? <SubmitConfirm /> : null}

            <div className='flex flex-wrap  justify-center space-x-4 lg:justify-start'>
                <div className='lg:basis-1/6 '></div>
                <img
                    className='my-3 md:w-[14rem] md:h-[14rem] rounded-full border-2 border-white '
                    src='data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOEAAADhCAMAAAAJbSJIAAAAhFBMVEX///8wMDAbGxvCwsIeHh4tLS0iIiKxsbGZmZknJycqKiolJSX6+vrp6ekhISFERETx8fHd3d1WVlbU1NRoaGgzMzOkpKT29vbj4+N1dXWBgYHIyMioqKhJSUliYmLq6uoTExOLi4tZWVl8fHw9PT2QkJAAAAAPDw9ubm7Nzc24uLhHR0fWtxrEAAAEMUlEQVR4nO3c23aiMBgF4HIIShA5gyCIoLbqvP/7DfQ47XhILCbEtb+b3mav5E/SJPj0BAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAC3mnWyLOv/yG7KPWRutMsrb/3y4lX75uD6shs0sIXRVKZlE6JpGiG2Y3rFMZDdqAH5Rjx3+nBfiGO2R1d2w4ayLBJb+x+lufEQBTkrK5ucCNixt9MHKEe/0c7k68eqlis/Ut3WOh+w70ZP8QnHr5xL+fpqTJTuxVk+vxKw68VQ5VpMrwfUNKfIZLfzZsftxRp8R7RSdkNvVe8pQ8CuFFtVZ5syYenCPuJUdlNvE6zYurAbp9VBdmNvUl5bKL6YjYqTjZuf2oyeGaZKVmLkMVZhz17Kbu4Njixr4Yf5VL1h6m/YB2k/TGvZDebmhqwzaY8kC9kN5uYy7Wc+rdVLuGBfK3q6eitisOZKaE5kN5iboXMlnKeyG8xtYnIltNTbmiIhEo4fEiLh+CEhEo4fEiLh+CEhEo4fEiLh+D1+wiXnSZR6CaOHP01c8CXUj7IbzI3zzHut3pn349/M+A1PJ9JYvdu1GddyYe3UuyF9iniu16iKt9w1xyUwXalXhp0J+1TjTJV8LLyIWTuRKvpiiP3VF1Fvy/bGZ+xEGqr4XujVgenREEnU25N+Ks+90v+X08hu5m8U18epFSs5j37Iimtvv6y9kkvhF7e4PFCtXPGA3YQ6vfRFid0o/S3Cu0ly7r8Mm6YKbrhPqDene5GoXoK9WebXdWSEpyIS73ioa99Xdyr1g2iZFvHWWptn+nCum9tVnhpRoOD/v8FyUoSOPrfp5TWfUHuuW1VeGoFCJekfyt1qa7LsZz5y2mayakpDjYnVaPaV5lzpulOd6ZAqbkb/VWlWrjzK0Xk/U9Jtm455gg2KLeHuvJ8hSbIf6aGNfwx153fx3kPaay8dX0X6xlbnOQK+EtIkk3EtINlhc/lz3xsyxsaIvi2Ndh7P5yNs7KQZy/mUn7a/nF5OI3a4G0U5HtpkuAL8jmrhCC7dyrvle8u4k5zP38zvMUC/ECuWugNY7K275utZq0hewGDPd9F7G6eVtseJ4vv3YM8ODTkBF62IHnyN+EfOQI1FBewiVjL2N4W4gF0tSjgYT8l9l4kfqPDLjai650L/P+IJnm38jdiAXSnuxe5RJ6zXu8MhqchSZL+iH47Ym2KOZxbDsQV2osv8uyxDIpW4SlzyvawcilmK6sQZw931PdBQ1MF/LX4ifaOLmms4f0xgOMIeg3P8eNCwaCUoId+vlgxJF5RQzkzaW4spRM7vDIYk6JuFku9rmCE5hZCEUzGnM6fYuZCEO5H/3H9HYyEJJe1oXhO2QhJKWw77fZuQhJNnXZZnMTPNU2TIshz7Sw0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAf4C+ppIAaVeUbkAAAAASUVORK5CYII='
                    alt='ProfilePic'
                />
                <div className='flex flex-col justify-start sm:basis-7/12 mt-5'>
                    <h2 className='ml-5 font-quicksand text-4xl font-semibold'>
                        {info.name}
                    </h2>
                    <p className='mx-5 mt-5 pt-5 font-SourceSansPro text-sm'>
                        {info.description}
                    </p>
                </div>
            </div>
            
            <div className='mx-5 pb-5'>
                <div className='grid-rows-10 grid-cols-6 gap-y-3 pl-4 lg:grid '>
                    <h1 className='text-[#457B9D] col-start-2 row-start-3 pt-7 font-quicksand text-4xl font-semibold sm:justify-self-start'>
                        Profile
                    </h1>
                    <p className='font-SourceSansPro text-md sm:col-start-2 sm:col-end-4 sm:row-start-4 sm:justify-self-start '>
                        Email: {info.email}
                    </p>
                    <p className='pb-3 font-SourceSansPro text-md sm:col-start-2 sm:col-end-4 sm:row-start-5 sm:justify-self-start'>
                        Phone: {info.phone}
                    </p>

                    <form onSubmit={handleSubmit} className='sm:col-start-2 sm:col-end-6 sm:row-start-6 sm:justify-self-start '>
                        <label
                            className='text-gray-900 dark:text-gray-300 mb-2 block text-md font-medium sm:relative sm:col-start-2 sm:col-end-3 sm:row-start-6 sm:justify-self-start'
                            htmlFor='default_size'
                        >
                            Upload Your CV
                        </label>
                        <input
                            onChange={(e)=> onInputChange(e)}
                            className='w-auto rounded text-gray-900 border-gray-300 dark:text-gray-400 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 mb-10  cursor-pointer border bg-[#F5F6F6] text-sm focus:outline-none sm:col-start-2 sm:col-end-3 sm:justify-self-start'
                            id='default_size'
                            type='file'
                            name='file'
                            accept="application/pdf"
                        />
                        <button
                            type='submit'
                            value='Submit'
                            className='border-[#457B9D] bg-[#457B9D] rounded mr-1 mb-1 flex-none  border-2 py-[0.7rem] px-1 text-xs text-white shadow outline-none transition-all duration-150 ease-linear hover:shadow-lg focus:outline-none sm:absolute sm:col-start-3 sm:col-end-4 sm:row-start-5 sm:justify-self-start'
                        >
                            Upload 
                        </button>
                    </form>

                    <div></div>
                </div>
            </div>
            {/* down to here */}

        </div>
    );
}

export default VolunteerProfile;
