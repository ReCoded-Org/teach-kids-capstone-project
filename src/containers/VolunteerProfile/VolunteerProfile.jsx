import React, { useState } from "react";
import VolunteerForm from "./VolunteerForm";
import Axios from 'axios';
import SubmitConfirm from "./SubmitConfirm";
import Navbar from "../../components/layout/Navbar/Navbar";
import Events from "../../components/EventCarousel/Events/Events";
import Footer from '../../components/layout/Footer/Footer';

function VolunteerProfile () {

  const [info, setInfo]= useState([]); //changed {} to []
  const [updated, setUpdated] = useState(false)

  function handleChange(e) {
    setInfo({...info, [e.target.name]: e.target.value})
  }

  function dataSender(e){
    Axios.put("http://localhost:8000/volunteer/1", info)
    .then((response)=> {
      // console.log(response.data)
      setInfo(() => response.data)
      // console.log('info after dataSender',info)
      setUpdated(true)
      setTimeout(()=> setUpdated(false), 2000)
    });
  }

    React.useEffect(()=> {
      Axios.get('http://localhost:8000/volunteer/1')
      .then(res => {
        setInfo(res.data)
        // console.log('info after useEffect', info)
      })
      .catch(err=> {
        console.log('err', err)
      })
    }, [])


  return (
    
    <div className='bg-gray font-quicksand'>
      <Navbar />
      <VolunteerForm handleChange={handleChange} dataSender={dataSender}/>

      {updated ? 
        <SubmitConfirm />
        :
        null
      }


      <div className="flex flex-wrap  space-x-4 sm:justify-start justify-center">
        <div className="sm:basis-1/6 "></div>
        <img className='sm:basis-1/6 rounded-full my-3 border-2 border-white sm:w-1/5 sm:h-1/5 h-1/3 w-1/3' src='data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOEAAADhCAMAAAAJbSJIAAAAhFBMVEX///8wMDAbGxvCwsIeHh4tLS0iIiKxsbGZmZknJycqKiolJSX6+vrp6ekhISFERETx8fHd3d1WVlbU1NRoaGgzMzOkpKT29vbj4+N1dXWBgYHIyMioqKhJSUliYmLq6uoTExOLi4tZWVl8fHw9PT2QkJAAAAAPDw9ubm7Nzc24uLhHR0fWtxrEAAAEMUlEQVR4nO3c23aiMBgF4HIIShA5gyCIoLbqvP/7DfQ47XhILCbEtb+b3mav5E/SJPj0BAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAC3mnWyLOv/yG7KPWRutMsrb/3y4lX75uD6shs0sIXRVKZlE6JpGiG2Y3rFMZDdqAH5Rjx3+nBfiGO2R1d2w4ayLBJb+x+lufEQBTkrK5ucCNixt9MHKEe/0c7k68eqlis/Ut3WOh+w70ZP8QnHr5xL+fpqTJTuxVk+vxKw68VQ5VpMrwfUNKfIZLfzZsftxRp8R7RSdkNvVe8pQ8CuFFtVZ5syYenCPuJUdlNvE6zYurAbp9VBdmNvUl5bKL6YjYqTjZuf2oyeGaZKVmLkMVZhz17Kbu4Njixr4Yf5VL1h6m/YB2k/TGvZDebmhqwzaY8kC9kN5uYy7Wc+rdVLuGBfK3q6eitisOZKaE5kN5iboXMlnKeyG8xtYnIltNTbmiIhEo4fEiLh+CEhEo4fEiLh+CEhEo4fEiLh+D1+wiXnSZR6CaOHP01c8CXUj7IbzI3zzHut3pn349/M+A1PJ9JYvdu1GddyYe3UuyF9iniu16iKt9w1xyUwXalXhp0J+1TjTJV8LLyIWTuRKvpiiP3VF1Fvy/bGZ+xEGqr4XujVgenREEnU25N+Ks+90v+X08hu5m8U18epFSs5j37Iimtvv6y9kkvhF7e4PFCtXPGA3YQ6vfRFid0o/S3Cu0ly7r8Mm6YKbrhPqDene5GoXoK9WebXdWSEpyIS73ioa99Xdyr1g2iZFvHWWptn+nCum9tVnhpRoOD/v8FyUoSOPrfp5TWfUHuuW1VeGoFCJekfyt1qa7LsZz5y2mayakpDjYnVaPaV5lzpulOd6ZAqbkb/VWlWrjzK0Xk/U9Jtm455gg2KLeHuvJ8hSbIf6aGNfwx153fx3kPaay8dX0X6xlbnOQK+EtIkk3EtINlhc/lz3xsyxsaIvi2Ndh7P5yNs7KQZy/mUn7a/nF5OI3a4G0U5HtpkuAL8jmrhCC7dyrvle8u4k5zP38zvMUC/ECuWugNY7K275utZq0hewGDPd9F7G6eVtseJ4vv3YM8ODTkBF62IHnyN+EfOQI1FBewiVjL2N4W4gF0tSjgYT8l9l4kfqPDLjai650L/P+IJnm38jdiAXSnuxe5RJ6zXu8MhqchSZL+iH47Ym2KOZxbDsQV2osv8uyxDIpW4SlzyvawcilmK6sQZw931PdBQ1MF/LX4ifaOLmms4f0xgOMIeg3P8eNCwaCUoId+vlgxJF5RQzkzaW4spRM7vDIYk6JuFku9rmCE5hZCEUzGnM6fYuZCEO5H/3H9HYyEJJe1oXhO2QhJKWw77fZuQhJNnXZZnMTPNU2TIshz7Sw0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAf4C+ppIAaVeUbkAAAAASUVORK5CYII=' alt='ProfilePic' />
        <div className='sm:basis-2/6 flex flex-col justify-start'>
          <h2 className='font-quicksand ml-5 text-4xl font-semibold'>{info.name}</h2>
          <p className="text-xs pt-5 mx-5 mt-5 font-SourceSansPro">{info.description}</p>
        </div>
        <div className="sm:basis-2/6"></div>
      </div>
{/* copy from here */}
<div className="pb-5 mx-5">
      <div className="sm:grid pl-4 grid-cols-6 grid-rows-10 gap-y-3 ">
          <h1 className="pt-7 col-start-2 row-start-3 sm:justify-self-start font-quicksand text-4xl font-semibold text-customGreen">Profile</h1>
          <p className="text-sm font-SourceSansPro sm:col-start-2 sm:col-end-4 sm:row-start-4 sm:justify-self-start ">Email: {info.email}</p>
          <p className="text-sm font-SourceSansPro sm:col-start-2 sm:col-end-4 sm:row-start-5 sm:justify-self-start pb-3">Phone: {info.phone}</p>

        <form className="sm:col-start-2 sm:col-end-6 sm:row-start-6 sm:justify-self-start ">
          <label className="sm:relative block mb-2 sm:col-start-2 sm:col-end-3 sm:row-start-6 sm:justify-self-start text-sm font-medium text-gray-900 dark:text-gray-300" htmlFor="default_size">Upload Your CV</label>
          <input className="mb-5 sm:col-start-2 sm:col-end-3 sm:justify-self-start w-full text-sm text-gray-900 bg-[#F5F6F6] border border-gray-300 cursor-pointer dark:text-gray-400 focus:outline-none dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400" id="default_size" type="file" />
          <button type="submit" value="Submit" className='sm:absolute rounded-none flex-none sm:col-start-4 sm:col-end-5 sm:row-start-5 sm:justify-self-start text-xs py-1 px-1 border-2 border-customGreen bg-customGreen text-white shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150'>Upload CV</button>

          </form>

      <div>
  
</div>


      </div>

      
      </div>      
      {/* down to here */}

      
      
      <Events />
      <Footer />
      {/* add the diagonal line at the bottom of the component */}
    </div>
  );
}

export default VolunteerProfile; 