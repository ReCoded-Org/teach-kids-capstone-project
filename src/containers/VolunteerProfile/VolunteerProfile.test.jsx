import React from "react";
import renderer from "react-test-renderer";
import VolunteerProfile from "./VolunteerProfile";
import { MemoryRouter } from "react-router-dom";

it("renders correctly", () => {
    const test = renderer
        .create(
            <MemoryRouter>
                <VolunteerProfile />
            </MemoryRouter>
        )
        .toJSON();
    expect(test).toMatchSnapshot();
});
