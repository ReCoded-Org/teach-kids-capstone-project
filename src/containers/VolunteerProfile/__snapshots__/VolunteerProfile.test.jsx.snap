// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-gray font-quicksand"
>
  <div
    className=" flex flex-wrap justify-end py-2 mr-5"
  >
    <div
      className="basis-8/10"
    />
    <button
      className="flex justify-end bg-customGreen text-white active:bh-customGreen font-bold uppercase text-sm px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
      onClick={[Function]}
      type="button"
    >
      Edit Profile
    </button>
  </div>
  <div
    className="flex flex-wrap  space-x-4 sm:justify-start justify-center"
  >
    <div
      className="sm:basis-1/6 "
    />
    <img
      alt="ProfilePic"
      className="sm:basis-1/6 rounded-full my-3 border-2 border-white w-1/6 h-1/6"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOEAAADhCAMAAAAJbSJIAAAAhFBMVEX///8wMDAbGxvCwsIeHh4tLS0iIiKxsbGZmZknJycqKiolJSX6+vrp6ekhISFERETx8fHd3d1WVlbU1NRoaGgzMzOkpKT29vbj4+N1dXWBgYHIyMioqKhJSUliYmLq6uoTExOLi4tZWVl8fHw9PT2QkJAAAAAPDw9ubm7Nzc24uLhHR0fWtxrEAAAEMUlEQVR4nO3c23aiMBgF4HIIShA5gyCIoLbqvP/7DfQ47XhILCbEtb+b3mav5E/SJPj0BAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAC3mnWyLOv/yG7KPWRutMsrb/3y4lX75uD6shs0sIXRVKZlE6JpGiG2Y3rFMZDdqAH5Rjx3+nBfiGO2R1d2w4ayLBJb+x+lufEQBTkrK5ucCNixt9MHKEe/0c7k68eqlis/Ut3WOh+w70ZP8QnHr5xL+fpqTJTuxVk+vxKw68VQ5VpMrwfUNKfIZLfzZsftxRp8R7RSdkNvVe8pQ8CuFFtVZ5syYenCPuJUdlNvE6zYurAbp9VBdmNvUl5bKL6YjYqTjZuf2oyeGaZKVmLkMVZhz17Kbu4Njixr4Yf5VL1h6m/YB2k/TGvZDebmhqwzaY8kC9kN5uYy7Wc+rdVLuGBfK3q6eitisOZKaE5kN5iboXMlnKeyG8xtYnIltNTbmiIhEo4fEiLh+CEhEo4fEiLh+CEhEo4fEiLh+D1+wiXnSZR6CaOHP01c8CXUj7IbzI3zzHut3pn349/M+A1PJ9JYvdu1GddyYe3UuyF9iniu16iKt9w1xyUwXalXhp0J+1TjTJV8LLyIWTuRKvpiiP3VF1Fvy/bGZ+xEGqr4XujVgenREEnU25N+Ks+90v+X08hu5m8U18epFSs5j37Iimtvv6y9kkvhF7e4PFCtXPGA3YQ6vfRFid0o/S3Cu0ly7r8Mm6YKbrhPqDene5GoXoK9WebXdWSEpyIS73ioa99Xdyr1g2iZFvHWWptn+nCum9tVnhpRoOD/v8FyUoSOPrfp5TWfUHuuW1VeGoFCJekfyt1qa7LsZz5y2mayakpDjYnVaPaV5lzpulOd6ZAqbkb/VWlWrjzK0Xk/U9Jtm455gg2KLeHuvJ8hSbIf6aGNfwx153fx3kPaay8dX0X6xlbnOQK+EtIkk3EtINlhc/lz3xsyxsaIvi2Ndh7P5yNs7KQZy/mUn7a/nF5OI3a4G0U5HtpkuAL8jmrhCC7dyrvle8u4k5zP38zvMUC/ECuWugNY7K275utZq0hewGDPd9F7G6eVtseJ4vv3YM8ODTkBF62IHnyN+EfOQI1FBewiVjL2N4W4gF0tSjgYT8l9l4kfqPDLjai650L/P+IJnm38jdiAXSnuxe5RJ6zXu8MhqchSZL+iH47Ym2KOZxbDsQV2osv8uyxDIpW4SlzyvawcilmK6sQZw931PdBQ1MF/LX4ifaOLmms4f0xgOMIeg3P8eNCwaCUoId+vlgxJF5RQzkzaW4spRM7vDIYk6JuFku9rmCE5hZCEUzGnM6fYuZCEO5H/3H9HYyEJJe1oXhO2QhJKWw77fZuQhJNnXZZnMTPNU2TIshz7Sw0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAf4C+ppIAaVeUbkAAAAASUVORK5CYII="
    />
    <div
      className="sm:basis-2/6 flex flex-col justify-start"
    >
      <h2
        className="font-quicksand ml-5 text-4xl font-normal"
      >
        <div
          className="font-bold"
        >
          Name of Volunteer: 
        </div>
      </h2>
      <p
        className="text-sm pt-5 ml-7 mt-5 font-SourceSansPro"
      >
        <span
          className="font-bold"
        >
          Description: 
        </span>
      </p>
    </div>
    <div
      className="sm:basis-2/6"
    />
  </div>
  <div
    className="sm:grid  grid-cols-4 grid-rows-8 gap-y-3 "
  >
    <h1
      className="pt-7 col-start-2 row-start-3 sm:justify-self-start font-quicksand text-4xl font-semibold text-customGreen"
    >
      Profile
    </h1>
    <p
      className="text-sm font-SourceSansPro sm:col-start-2 sm:col-end-4 sm:row-start-4 sm:justify-self-start "
    >
      <span
        className="font-bold"
      >
        Email: 
      </span>
    </p>
    <p
      className="text-sm font-SourceSansPro sm:col-start-2 sm:col-end-4 sm:row-start-5 sm:justify-self-start pb-3"
    >
      <span
        className="font-bold"
      >
        Phone Number: 
      </span>
    </p>
  </div>
  <div
    className="flex justify-center pb-10"
  >
    <form
      className="  "
    >
      <input
        accept="application/pdf,application/vnd.ms-excel"
        className="flex-none text-sm  bg-gray"
        name="upload"
        type="file"
      />
      <input
        className="flex-none text-xs py-1 mx-2 my-3 px-10 border-2 border-customGreen bg-customGreen text-white rounded-md shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
        type="submit"
      />
    </form>
  </div>
</div>
`;
